Source/Core/Core/IOS/USB/USB_HID/HIDv4.cpp

```diff
@@ -172,15 +172,21 @@ void USB_HIDv4::OnDeviceChange(ChangeEvent event, std::shared_ptr<USB::Device> d
     if (event == ChangeEvent::Inserted)
     {
       s32 new_id = 0;
-      while (m_ios_ids.contains(new_id))
+      decltype(m_ios_ids)::iterator it;
+      while ((it = m_ios_ids.find(new_id)) != m_ios_ids.end())
         ++new_id;
-      m_ios_ids[new_id] = device->GetId();
-      m_device_ids[device->GetId()] = new_id;
+
+      const auto id = device->GetId();
+      it->second = id;
+      m_device_ids[id] = new_id;
```

Comment:
Pretty sure this isn't right? The old code implicitly inserts `new_id`, the new code dereferences the end iterator.